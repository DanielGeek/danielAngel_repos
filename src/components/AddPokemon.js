import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddPokemon = ({ setPokemones }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value.toLowerCase());
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 0) {
      setPokemones( pokemones => {
        const newpokemones = pokemones.filter( pokemon => pokemon !== inputValue)
        return [ inputValue, ...newpokemones]
      });
      setInputValue('');
    }

  }

  return (
    <>
      <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
        <input
          type="text"
          placeholder='Buscar'
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </>
  )
}

AddPokemon.propTypes = {
  setPokemones: PropTypes.func.isRequired
}
