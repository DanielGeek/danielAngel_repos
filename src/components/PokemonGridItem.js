import React from 'react';
import PropTypes from 'prop-types';
import { capitalizeFirstLetter } from '../helpers/capitalizeFirstLette';

export const PokemonGridItem = ({ name, url, id }) => {

  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={ url } alt={ name } />
      <p># { id }</p>
      <p>{ capitalizeFirstLetter(name) }</p>
    </div>
  )
}

PokemonGridItem.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
}
