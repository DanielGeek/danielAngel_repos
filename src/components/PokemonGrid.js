import React from 'react'
import PropTypes from 'prop-types';

import { useFetchPokemon } from '../hooks/useFetchPokemon'
import { PokemonGridItem } from './PokemonGridItem';

export const PokemonGrid = ({ pokemonNameOrId }) => {

  const { data: images, loading, error } = useFetchPokemon( pokemonNameOrId );

  let imagesFilters = images.filter(image => image.id !== pokemonNameOrId);

  return (
    <>

      { loading && <p className='animate__animated animate__flash'>Cargando</p>}
      { !error &&
          <div className='card-grid'>
          {
            imagesFilters.map( img => (
              <PokemonGridItem
                  key={ img.id }
                  { ...img }
              />
            ))
          }
        </div>
      }
    </>
  )
}

PokemonGrid.propTypes = {
  pokemonNameOrId: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
}