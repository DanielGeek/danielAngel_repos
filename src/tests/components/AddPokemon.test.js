import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { AddPokemon } from "../../components/AddPokemon";

describe('Pruebas en <AddPokemon />', () => {

  const setPokemones = jest.fn();
  let wrapper = shallow(<AddPokemon setPokemones={setPokemones} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddPokemon setPokemones={setPokemones} />);
  });

  test('debe de mostrarse correctamente', () => {

    expect(wrapper).toMatchSnapshot();

  });

  test('debe de cambiar la caja de texto', () => {

    const input = wrapper.find('input');
    const value = '1';

    input.simulate('change', { target: { value }});

    expect(wrapper.find('p').text().trim()).toBe(value);

  });

  test('No debe de postear la informaciÃ³n con submit', () => {

    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(setPokemones).not.toHaveBeenCalled()

  });

  test('debe de llamar el setPokemones y limpiar la caja de texto', () => {

    const value = '1';

    wrapper.find('input').simulate('change', { target: { value }});

    wrapper.find('form').simulate('submit', { preventDefault(){}});

    expect(setPokemones).toHaveBeenCalled();
    expect(setPokemones).toHaveBeenCalledTimes(1);
    expect(setPokemones).toHaveBeenCalledWith(expect.any(Function));

    expect(wrapper.find('input').prop('value')).toBe('');
  });
});