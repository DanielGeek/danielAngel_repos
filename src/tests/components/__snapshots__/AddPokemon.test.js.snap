// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddPokemon /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      placeholder="Buscar"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
